pyteal/ast/opup.py

```diff
@@ -13,6 +14,8 @@
 from pyteal.types import TealType, require_type
 from enum import Enum
 
+ON_CALL_APP = Bytes("base16", "068101")  # v6 program "int 1"
```

Comment:
Maybe for a follow up pr but should we make this version take the program version from the compile options?

beaker has a half baked concept of a `Precompile` for things like this https://github.com/algorand-devrel/beaker/blob/boxen/beaker/precompile.py

Reply:
Currently I don't think there's a benefit to having this inner program version match the outer program version. However if we ever make it so that a new AVM version cannot call into v6 programs, this definitely must be changed.